src/Policies/Policy.php

```diff
@@ -1,34 +1,39 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Mazedlx\FeaturePolicy\Policies;
 
 use Illuminate\Http\Request;
 use Mazedlx\FeaturePolicy\FeatureGroups\DefaultFeatureGroup;
+use Mazedlx\FeaturePolicy\Formatter\PolicyFormatter;
 use Mazedlx\FeaturePolicy\Value;
 use Mazedlx\FeaturePolicy\Directive;
-use Stringable;
 use Symfony\Component\HttpFoundation\Response;
 
-abstract class Policy implements PolicyContract, Stringable
+abstract class Policy implements PolicyContract
 {
     protected array $directives = [];
 
+    private array $rules = [];
+
     abstract public function configure();
 
     public function addDirective(string $directive, $values, ?string $type = DefaultFeatureGroup::class): self
     {
+        $this->rules[$directive] ??= [];
         $currentDirective = Directive::make($directive, type: $type);
+        collect($this->rules[$directive])
+            ->each(fn (string $rule) => $currentDirective->addRule($rule));
```

Comment:
get existing rules and load them into the current directive
```suggestion
        $currentDirective = Directive::make($directive, type: $type);
        collect($this->rules[$directive] ??= [])
            ->each(fn (string $rule) => $currentDirective->addRule($rule));
```